src/SchedulerSubsystem/Scheduler.java

```diff
@@ -72,4 +114,51 @@ public void run() {
             }
         }
     }
+
+    /* Test */
+
+    /**
+     * Nested test class for class Scheduler. Contains unit tests for critical private methods.
+     *
+     * See README section Test for additional context.
+     */
+    public static class SchedulerTest {
```

Comment:
I don't have time to research if this is good or not, I'm not reviewing this part.